include/config.hpp

```diff
@@ -0,0 +1,4 @@
+#pragma once
```

Comment:
we wszystkich innych miejscach korzystasz z ifndef define. Wiec tutaj też bym tak zrobił.

`#pragma once` z tego pamiętam nie jest standardem - tj. nie każdy kompilator c++ ci to obsłuży (co nie oznacza ze jest złe)